### PR TITLE
Don't update bg images data when new images are not fetched yet

### PR DESCRIPTION
**Pull request by sangwoo108**
Resolves https://github.com/brave/brave-browser/issues/26135

We should use the default image until we get new images

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_new_tab_ui/data/backgrounds.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ export const images: NewTab.BraveBackground[] = [{
 // then you must also update `script/generate_licenses.py`
 
 export const updateImages = (newImages: NewTab.BraveBackground[]) => {
-  if (!newImages) {
+  if (!newImages?.length) {
```
Comment (sangwoo108) [inline components/brave_new_tab_ui/data/backgrounds.ts:17]: Obviously, I made a mistake here. An empty [] is evaluated true.

Comment (simonhong) [reply components/brave_new_tab_ui/data/backgrounds.ts:17]: why don't use `!newImages.length` here if empty array is passed?

Comment (sangwoo108) [reply components/brave_new_tab_ui/data/backgrounds.ts:17]: Then the default image will be gone.

Comment (sangwoo108) [reply components/brave_new_tab_ui/data/backgrounds.ts:17]: Yeah, it's kinda confusing. Array.splice() will remove from the 0 to image.length count, and then insert given items to the array. So, if newImages are empty, it'll only remove all existing items and not insert new items.

Comment (sangwoo108) [reply components/brave_new_tab_ui/data/backgrounds.ts:17]: Aha, I was mistaken about your comment. Yeah, we can remove "?". newImages can't be undefined or null so we don't need it.

Comment (sangwoo108) [reply components/brave_new_tab_ui/data/backgrounds.ts]: Removed! Thanks :)

**Review by simonhong — APPROVED** [review]
++ 👍🏼

**Comment by sangwoo108** [comment]
Thank you!

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.47.10 Chromium: 107.0.5304.62 (Official Build) nightly (64-bit)
-- | --
Revision | 1eec40d3a5764881c92085aaee66d25075c159aa-refs/branch-heads/5304@{#942}
OS | Windows 11 Version 22H2 (Build 22621.675)
```

Went through the STR/Cases outlined via https://github.com/brave/brave-browser/issues/26135#issue-1415325749 and ensured that the default NTP image was being displayed rather than the gradient colour as per the following:

https://user-images.githubusercontent.com/2602313/197457415-b4687dc6-81e0-4aca-8d9d-2820a8783e81.mp4

**Pull request merged**